pkg/config/labels/instance.go

```diff
@@ -98,6 +100,69 @@ func IsDNS1123Label(value string) bool {
 	return len(value) <= DNS1123LabelMaxLength && dns1123LabelRegexp.MatchString(value)
 }
 
+func isAlpha(c rune) bool {
+	return c >= 'a' && c <= 'z'
+}
+
+func isDigit(c rune) bool {
+	return c >= '0' && c <= '9'
+}
+
+// ToDNS1123Label attempts to convert the given value to conform to the definition of a label in
+// DNS (RFC 1123).
+func ToDNS1123Label(value string) (string, error) {
```

Comment:
Nit: 
K8S DNS Label name has a length limit: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names

AFAIK, some full resource name can potentially exceed that limit. To comply, maybe we should use DNS Subdomain name vs label name and allow `.`.